### PR TITLE
Implement Triggers store in Zustand

### PR DESCRIPTION
**Pull request by pavloburchak**
This PR...

## Changes
- https://github.com/kubeshop/testkube/issues/3765

## Fixes

-

## How to test it

-

## screenshots

-

## Checklist

- [ ] tested locally
- [ ] added new dependencies
- [ ] updated the docs
- [ ] added a test

### CODE REVIEW

File: src/store/LocalStore.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import {create} from 'zustand';
+import {devtools} from 'zustand/middleware';
+import {shallow} from 'zustand/shallow';
+
+import createTriggersSlice, {TriggersSlice} from './TriggersSlice';
+
+/* Create store. For more information,
+check https://github.com/pmndrs/zustand/blob/main/docs/guides/typescript.md#basic-usage  */
+const useLocalStore = create<TriggersSlice>()(
```
Comment (rangoo94) [inline src/store/LocalStore.ts:9]: It is a global store, yet it should be local = without global space pollution. @haneabogdan, could you assist with that, based on your findings?

File: src/store/TriggerLocalStore.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+          )
+    );
+    const useStore = useStoreRef.current;
+          
+    const useShallowLocalStore = <U>(selector: (state: TriggersSlice) => U) => useStore(selector, shallow);
+
+    return [useShallowLocalStore];
+}; 
+
+export default useTriggerLocalStore;
```
Comment (haneabogdan) [inline src/store/TriggerLocalStore.ts:25]: this is a draft implementation of local store.
Usage would be:

`  const [useShallowLocalStore] = useTriggerLocalStore();
// whatever props we want to fetch from store
  const {text, setText} = useShallowLocalStore(state => ({
    text: state.text,
    setText: state.setText,
  }));`

Comment (haneabogdan) [reply src/store/TriggerLocalStore.ts:25]: Tested it locally with 2 instances of the same store on different components

Comment (rangoo94) [reply src/store/TriggerLocalStore.ts:25]: Nice one! Is there a way to provide the initial state?

Comment (pavloburchak) [reply src/store/TriggerLocalStore.ts:25]: I believe it is provided in TriggersSlice itself, isn't is @haneabogdan ?

**Review by haneabogdan — APPROVED** [review]

**Pull request merged**